mlir/include/mlir/Dialect/Rock/IR/TransformMapBuilder.h

```diff
@@ -44,8 +44,8 @@ class TransformMapBuilder {
   void getEndNames(SmallVectorImpl<StringRef> &names);
   void getStartNames(SmallVectorImpl<StringRef> &names);
 
-  SmallString<8> startName(uint32_t dim);
-  SmallString<8> endName(uint32_t dim);
+  StringRef startName(uint32_t dim);
```

Comment:
BottomUpTMBuilder::broadcast() called startName() to acquire a string which it saved in a vector of StringRef.  The call constructed a temp SmallString for the return value, which went away when the scope was exited, leaving the StringRef dangling.  It appears to me that it didn't really want a string copy, just a pointer into TransformMapBuilder's startNames collection.

Reply:
Yeah, reasonable